edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/DependencyManager.cs

```diff
@@ -88,8 +87,16 @@ public void Register(ContainerBuilder builder)
 
             MetricsConfig metricsConfig = new MetricsConfig(this.configuration.GetSection("metrics:listener"));
 
-            this.RegisterCommonModule(builder, optimizeForPerformance, storeAndForward, metricsConfig);
-            this.RegisterRoutingModule(builder, storeAndForward, experimentalFeatures);
+            if (!Enum.TryParse(this.configuration.GetValue("AuthenticationMode", string.Empty), true, out AuthenticationMode authenticationMode))
+            {
+                authenticationMode = AuthenticationMode.Scope;
+            }
+
+            bool trackDeviceState = authenticationMode == AuthenticationMode.Scope
+                && this.configuration.GetValue("TrackDeviceState", true);
```

Comment:
Trying to think what would be the most intuitive and future proof env var name here.. 
naming is hard :)

Reply:
I think TrackDeviceState might be too generic, maybe something like MessagesDropWhenDeviceInvalidState, MessagesDropWhenOutOfScope or MessagesHandlingWhenDeviceInvalidState and have an enum of Drop, keep. Also should we use State or Status? I think Status is more appropiate.

Reply:
Yes, I agree. But in a way TrackDeviceState/Status makes it more generic, in case we want to add more changes to how device deletions are handled (say remove twin/DMs/Subs, etc)